ironic/drivers/modules/drac/management.py

```diff
@@ -308,7 +331,143 @@ class DracRedfishManagement(redfish_management.RedfishManagement):
     specific incompatibilities and introduction of vendor value added
     should be implemented by this class.
     """
-    pass
+    @METRICS.timer('DracRedfishManagement.clear_job_queue')
+    @base.clean_step(priority=0)
+    def clear_job_queue(self, task):
+        """Clear the job queue.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: DracOperationError on an error from python-dracclient.
+        """
+        self.get_redfish_info(task)
+        jobs_url = 'https://%s%s' % (self.redfish_ip, _JOBS_BASE_URL)
+        job_response = requests.get(
+            jobs_url,
+            auth=(self.redfish_username, self.redfish_password), verify=False)
+
+        jobs_data = job_response.json()
+        job_list = re.findall("JID_.+?'", str(jobs_data))
+        if job_list is None:
+            LOG.info(
+                'Job queue already cleared for iDRAC %s'
+                % (self.redfish_ip))
+            return None
+
+        try:
+            for job_id in job_list:
+                job_id = job_id.strip("'")
+                job_url = '%s/%s' % (jobs_url, job_id)
+                delete_response = requests.delete(
```

Comment:
It looks like the job queue can be cleared in 1 call, without having to loop over the job IDs by passing in "JID_CLEARALL" for the job ID.  See: https://github.com/dell/iDRAC-Redfish-Scripting/blob/cbff04525e97e29b2714f9d91b6fbacc335bbd46/Redfish%20Python/GetDeleteJobQueueREDFISH.py#L31
and:
https://github.com/dell/iDRAC-Redfish-Scripting/blob/cbff04525e97e29b2714f9d91b6fbacc335bbd46/Redfish%20Python/GetDeleteJobQueueREDFISH.py#L84
Also, we need to check the http error code returned and raise an exception if the clear failed.

Reply:
Addressed by passing job ID's as "JID_CLEARALL".